### ISSUE TITLE
[Improvement][Medium] Isolate Brax/MJX compatibility hacks and private API dependencies

### ISSUE THREAD

**Issue by MarcusFunt** [MarcusFunt/Ascento_MuJoCo_Playground#27]
## Type
Improvement / maintainability

## Priority
**Medium**

## Problem
Canonical training/environment code currently depends on several version-sensitive implementation details:

- monkey-patching `jax.device_put_replicated`;
- monkey-patching Brax PPO gradient/loss functions in `training/train.py`;
- using private `distribution._NormalDistribution` in `training/bounded_distribution.py`;
- reading `data._impl.cfrc_ext` in `ascento/observations.py`.

Dependencies are pinned, which reduces immediate risk, but these assumptions are spread across core modules and are easy to miss during upgrades.

## Recommended implementation
Create a dedicated compatibility layer, for example:

```text
ascento/compat/
  __init__.py
  brax_013.py
  mjx_335.py
```

Responsibilities:

- assert supported JAX/Brax/MuJoCo versions;
- apply any required monkey patches in one place;
- wrap private API access behind documented helper functions;
- provide a small compatibility smoke test.

Where reasonable, replace private APIs with public equivalents.

Longer term, if PPO customization continues to grow, consider owning a small local trainer wrapper rather than patching upstream Brax globals at import time.

## Acceptance criteria
- [ ] Version-sensitive patches are centralized.
- [ ] Supported dependency versions are checked explicitly at startup/test time.
- [ ] Core environment code does not directly reach into MJX private state without a compatibility wrapper.
- [ ] A dependency upgrade produces a clear compatibility failure rather than subtle runtime behavior changes.

**Comment by MarcusFunt** [comment]
Closing as superseded by the merged mjlab/RSL-RL migration (PR #51). This issue targets the removed legacy JAX/MJX/custom-training implementation. The current repository now uses the mjlab managers and RSL-RL workflow; any remaining requirement should be reopened as a new issue against the current architecture.
